main/driver.c

```diff
@@ -2105,9 +2109,9 @@ static void settings_changed (settings_t *settings, settings_changed_flags_t cha
 
         uint32_t i = sizeof(inputpin) / sizeof(input_signal_t);
 
-        do {
+        while(i--) {
```

Comment:
This change will lead to i = 0 not beeing processed, at least with the compiler used by ST. Is that your intention?

Reply:
Yes, if we do not have configured input pins, then we shouldn't enter this loop. Otherwise, at this particular place we get a bug at line 2110. In other places I get a compiler warning.
btw, need to change i initialization (line 2106) to inputPinsCounts - I missed that.

Reply:
My point was that the loop is not executed when i == 0, it should when i > 0 initially. The correct way is to test for i > 0 before entering the loop like this:
```
if(i) do {
    i--;
    ...
} while(i);
```
or use a for loop instead - which I avoid if I can due to beeing an old hand that started programming in assembler...

Reply:
the code:
`while(i--) {}`
is identical to:
`if(i) do{} while(i)`
if i is unsigned, and its value not used afterwards.
But I'm ok with your option. Do you want me to rework fix in this way?

Reply:
> Do you want me to rework fix in this way?

I think it is clearer the way I do it, and it is how I do it elsewhere. You chose.

Reply:
Ok